source/dub/project.d

```diff
@@ -354,7 +354,10 @@ class Project {
 					}
 				} else if (m_selections.hasSelectedVersion(basename)) {
 					vspec = m_selections.getSelectedVersion(basename);
-					if (vspec.path.empty) p = m_packageManager.getBestPackage(dep.name, vspec);
+					if (!vspec.repository.empty) {
+						p = m_packageManager.clonePackage(basename, vspec);
+						if (subname.length) p = m_packageManager.getSubPackage(p, subname, true);
```

Comment:
this getSubPackage line is now copy-pasted 5 times into this function, let's add a function inside this function which does the if check + assignment for us, such as `resolveSubPackage(p, true);`